modules/renter/hostdb/negotiate.go

```diff
@@ -377,38 +390,39 @@ func (hu *hostUploader) revise(rev types.FileContractRevision, piece []byte, hei
 }
 
 // newHostUploader negotiates an initial file contract with the specified host
-// and returns a hostUploader, which satisfies the uploader interface.
-func (r *Renter) newHostUploader(settings modules.HostSettings, filesize uint64, duration types.BlockHeight, masterKey crypto.TwofishKey) (*hostUploader, error) {
+// and returns a hostUploader, which satisfies the Uploader interface.
+func (hdb *HostDB) newHostUploader(settings modules.HostSettings) (*hostUploader, error) {
 	// reject hosts that are too expensive
 	if settings.Price.Cmp(maxPrice) > 0 {
 		return nil, errTooExpensive
 	}
 
 	hu := &hostUploader{
-		settings:  settings,
-		masterKey: masterKey,
-		tree:      crypto.NewTree(),
-		renter:    r,
+		settings: settings,
+		tree:     crypto.NewTree(),
+		hdb:      hdb,
 	}
 
 	// get an address to use for negotiation
 	// TODO: use more than one shared address
-	if r.cachedAddress == (types.UnlockHash{}) {
-		uc, err := r.wallet.NextAddress()
+	if hdb.cachedAddress == (types.UnlockHash{}) {
+		uc, err := hdb.wallet.NextAddress()
 		if err != nil {
 			return nil, err
 		}
-		r.cachedAddress = uc.UnlockHash()
+		hdb.cachedAddress = uc.UnlockHash()
 	}
 
-	// TODO: check for existing contract?
-	err := hu.negotiateContract(filesize, duration, r.cachedAddress)
+	const filesize = 1 << 32 // 4 GiB
+	const duration = 6000    // 6 weeks
```

Comment:
4 GiB is arbitrary. Not sure how big these contracts should be. I suppose eventually we will want them to grow larger/smaller as needed.